src/push.rs

```diff
@@ -99,12 +99,58 @@ pub async fn push_profile(data: PushProfileData<'_>) -> Result<(), PushProfileEr
         Command::new("nix-build")
     };
 
+    let store_address = format!("ssh-ng://{}@{}",
+        if data.deploy_data.profile.generic_settings.ssh_user.is_some() {
+            &data.deploy_data.profile.generic_settings.ssh_user.as_ref().unwrap()
+        } else {
+            &data.deploy_defs.ssh_user
+        },
+        data.deploy_data.node.node_settings.hostname
+    );
+
+    let ssh_opts_str = data
+        .deploy_data
+        .merged_settings
+        .ssh_opts
+        // This should provide some extra safety, but it also breaks for some reason, oh well
+        // .iter()
+        // .map(|x| format!("'{}'", x))
+        // .collect::<Vec<String>>()
+        .join(" ");
+
+
     if data.supports_flakes {
         build_command.arg("build").arg(derivation_name)
     } else {
         build_command.arg(derivation_name)
     };
 
+    // copy derivation over to remote server
+    if data.supports_flakes && data.deploy_data.merged_settings.remote_build.unwrap_or(false) {
+        info!("Building on remote host");
+
+        // copy the derivation to remote host so it can be built there
+        let copy_command_status = Command::new("nix").arg("copy")
+            .arg("-s")  // fetch dependencies from substitures, not localhost
+            .arg("--to").arg(&store_address)
+            .arg("--derivation").arg(derivation_name)
+            .env("NIX_SSHOPTS", ssh_opts_str.clone())
```

Comment:
Okay, it doesn't seem to work that easily :(

```
🚀 ℹ️ [deploy] [INFO] Building profile `system` for node `<bla>`
🚀 ℹ️ [deploy] [INFO] Building on remote host
ssh: connect to host <bla-bla> port 22: Network is unreachable
error: cannot open connection to remote store 'ssh-ng://rvem@<bla-bla>': error: unexpected end-of-file
🚀 ❌ [deploy] [ERROR] Failed to push profile: Nix build command resulted in a bad exit code: Some(1)
```

Reply:
bThough, according to debug logs OpenSSH succesfully  the non-default port:
```
Authenticated to <bla-bla> ([<bla-bla IP>]:17788) using "publickey".
```
But then
```
debug1: Sending command: nix-daemon --stdio
copying 0 paths...
starting pool of 15 threads
killing process 421682
kex_exchange_identification: read: Connection reset by peer
Connection reset by <bla-bla IP> port 22
```

Reply:
Oh, looks like it fails somewhere further